### ISSUE TITLE
Feature Request: webhook for notifications

### ISSUE THREAD

**Issue by J-Run** [nesfit/fitcrack#54]
It would be great to have a webhook functionality that make possible to send valueable events notifications such as:
- N new hashes cracked on job M
- job M finished
- An error has occurred on job M

to telegram chat with bot, or any other notification api that accepts GET req with params.

Example webhook URI for telegram:
`https://api.telegram.org/bot000000000:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/sendMessage?chat_id=-0000000000000&text=URL_encoded_message_text`

**Comment by davidbolvansky** [comment]
Yeah, it sounds as a good idea. But not sure how it could look like, simply "some_user_defined_url&text=message_from_fitcrack"?

So for your use case, you would enter:
"https://api.telegram.org/bot000000000:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/sendMessage?chat_id=-0000000000000&text="

?

**Comment by J-Run** [comment]
Yeah, but i believe that not only telegram is widely used for notification purposes so some kind of template is applicable here.
e.g.:
```
https://api.telegram.org/bot000000000:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/sendMessage?chat_id=-0000000000000&text={fitcrack_message}
```
where substring `{fitcrack_message}` will be replaced with actual urlencoded message

**Comment by felixd** [comment]
Check Apprise: https://github.com/caronc/apprise 

That might help a bit ;)

**Comment by davidbolvansky** [comment]
Apprise looks really nice
